patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch

```diff
@@ -1,12 +1,12 @@
 diff --git a/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc b/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
-index f6ef61a0ac86ae55e8f9e80ece0758253af48520..888d656cf7080ebc32e90fc492e70a9b394d7026 100644
+index f6ef61a0ac86ae55e8f9e80ece0758253af48520..d4e435dbda3b1a26690b33c73cfb21bdba3b6cdd 100644
 --- a/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
 +++ b/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
 @@ -67,6 +67,7 @@ bool BookmarkTabHelper::ShouldShowBookmarkBar() const {
        !prefs->GetBoolean(bookmarks::prefs::kShowBookmarkBar))
      return false;
  
-+  return false;
++  ReturnFalseIfBookmarkBarShouldHide(profile);
```

Comment:
This patch is inevitable because we have to return in the middle of existing logic.

Reply:
there is already a NTP check below. If we need to change the behavior of that check we can do a chromium_src override of `search::NavEntryIsInstantNTP`

Reply:
I guess I'm not sure if the override will work because of the `||` and I'm not sure what the real url is for a custom NTP, but let's discuss in dm

Reply:
another option here is to override `BookmarkTabHelper` in browser.cc and replace with our own tab helper that proxies to the original instead of subclassing it

Reply:
`BraveBookmarkTabHelper` is introduced to proxy `BookmarkTabHelper`.